Description: Show a default icon for .desktop files that don't have one
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=76495

Index: gnome-panel-2.30.0/gnome-panel/menu.c
===================================================================
--- gnome-panel-2.30.0.orig/gnome-panel/menu.c	2010-06-03 13:57:54.293946027 +1000
+++ gnome-panel-2.30.0/gnome-panel/menu.c	2010-06-03 13:57:57.336455621 +1000
@@ -1491,7 +1491,7 @@
 					NULL, NULL,
 					alias_directory ? gmenu_tree_directory_get_icon (alias_directory) :
 							  gmenu_tree_entry_get_icon (entry),
-					NULL);
+					PANEL_STOCK_DEFAULT_ICON);
 
 	setup_menuitem (menuitem,
 			panel_menu_icon_get_size (),
Index: gnome-panel-2.30.0/gnome-panel/panel-stock-icons.h
===================================================================
--- gnome-panel-2.30.0.orig/gnome-panel/panel-stock-icons.h	2010-01-14 14:26:27.000000000 +1100
+++ gnome-panel-2.30.0/gnome-panel/panel-stock-icons.h	2010-06-03 13:57:57.336455621 +1000
@@ -46,6 +46,8 @@
 #define PANEL_STOCK_EXECUTE             "panel-execute"
 #define PANEL_STOCK_CLEAR               "panel-clear"
 #define PANEL_STOCK_DONT_DELETE         "panel-dont-delete"
+/* FIXME: put a more representative icon here */
+#define PANEL_STOCK_DEFAULT_ICON		"application-default-icon"
 
 void        panel_init_stock_icons_and_items (void);
 GtkIconSize panel_menu_icon_get_size         (void);
